test/gtest/ucp/test_ucp_tag_xfer.cc

```diff
@@ -1242,41 +1285,75 @@ class multi_rail_max : public test_ucp_tag_xfer {
-    test_run_xfer(true, true, true, true, false);
+        DIR *dir = opendir(ibdev_sysfs_dir);
+        if (dir == NULL) {
+            UCS_TEST_SKIP_R(std::string(ibdev_sysfs_dir) + " not found");
+        }
 
-    auto ep_num_lanes = ucp_ep_num_lanes(sender().ep());
-    ASSERT_EQ(ep_num_lanes, ucp_ep_num_lanes(receiver().ep()));
+        for (;;) {
+            struct dirent *entry = readdir(dir);
+            if (entry == NULL) {
+                break;
+            }
 
-    if (is_proto_enabled()) {
-        EXPECT_EQ(num_lanes(), ep_num_lanes);
-    }
+            if (entry->d_name[0] == '.') {
+                continue;
+            }
 
-    size_t bytes_sent = 0;
-    unsigned num_used_lanes = 0;
-    for (ucp_lane_index_t lane = 0; lane < ep_num_lanes; ++lane) {
-        size_t sender_tx   = get_bytes_sent(sender().ep(), lane);
-        size_t receiver_tx = get_bytes_sent(receiver().ep(), lane);
-        UCS_TEST_MESSAGE << "lane[" << static_cast<int>(lane) << "] : "
-                         << "sender " << sender_tx << " receiver " << receiver_tx;
-        if ((sender_tx > 0) || (receiver_tx > 0)) {
-            ++num_used_lanes;
+            devices.push_back(entry->d_name);
         }
-        bytes_sent += sender_tx + receiver_tx;
+
+        closedir(dir);
     }
 
-    /* One lane could be reserved for tag offload and not selected for AM/RMA
-       bandwidth lane */
-    EXPECT_GE(num_used_lanes, ep_num_lanes - 1);
+    void test_max_lanes(int num_paths) {
+        receiver().connect(&sender(), get_ep_params());
+        test_run_xfer(true, true, true, true, false);
+
+        auto ep_num_lanes = ucp_ep_num_lanes(sender().ep());
+        ASSERT_EQ(ep_num_lanes, ucp_ep_num_lanes(receiver().ep()));
+
+        if (is_proto_enabled()) {
+            EXPECT_EQ(num_paths, ep_num_lanes);
+        }
 
-    EXPECT_GE(bytes_sent, get_msg_size());
+        size_t bytes_sent = 0;
+        unsigned num_used_lanes = 0;
+        for (ucp_lane_index_t lane = 0; lane < ep_num_lanes; ++lane) {
+            size_t sender_tx   = get_bytes_sent(sender().ep(), lane);
+            size_t receiver_tx = get_bytes_sent(receiver().ep(), lane);
+            UCS_TEST_MESSAGE << "lane[" << static_cast<int>(lane) << "] : "
+                            << "sender " << sender_tx << " receiver " << receiver_tx;
+            if ((sender_tx > 0) || (receiver_tx > 0)) {
+                ++num_used_lanes;
+            }
+            bytes_sent += sender_tx + receiver_tx;
+        }
+
+        /* One lane could be reserved for tag offload and not selected for AM/RMA
+        bandwidth lane */
+        EXPECT_GE(num_used_lanes, ep_num_lanes - 1);
+
+        EXPECT_GE(bytes_sent, get_msg_size());
+    }
+};
+
+UCS_TEST_P(multi_rail_max, max_lanes, "TM_SW_RNDV=y",
+           "RNDV_THRESH=1", "MIN_RNDV_CHUNK_SIZE=1", "MULTI_PATH_RATIO=0.0001")
+{
+    if (get_variant_value() == VARIANT_64_PATHS) {
+        test_max_lanes(64);
```

Comment:
instead of variants we can just another test case with UCS_TEST_P - one for 64, another of 128